modules/styling/lib/cs.ts

```diff
@@ -489,7 +489,7 @@ export function handleCsProp<
   // _does_ care because it uses the `classList` order to determine style merging. Therefore we
   // should always be careful to order class names knowing we need to support runtime merging if
   // runtime styles are detected.
-  const returnProps = csToProps([localCs, className, cs, style]);
+  const returnProps = csToProps([localCs, className, cs]);
```

Comment:
Adding `style` to the `csToProps` was always weird. But before we added support for static inline styles, it worked. Afterwards, `csToProps` couldn't differentiate between a `style` prop given from `elemProps` or one given by a `createVars` function or a Stencil. So I removed `style` from the array and merge it below instead.